src/lib/routes/admin-api/metrics.ts

```diff
@@ -188,15 +188,31 @@ class MetricsController extends Controller {
         res: Response<ApplicationsSchema>,
     ): Promise<void> {
         const { user } = req;
-        const query = req.query.strategyName
-            ? { strategyName: req.query.strategyName as string }
-            : {};
+        const { query, offset, limit = '1000', sortOrder, sortBy } = req.query;
+
+        const normalizedQuery = query
+            ?.split(',')
+            .map((query) => query.trim())
+            .filter((query) => query);
+
+        const normalizedLimit =
+            Number(limit) > 0 && Number(limit) <= 1000 ? Number(limit) : 1000;
+        const normalizedOffset = Number(offset) > 0 ? Number(offset) : 0;
+        const normalizedSortBy: string = sortBy ? sortBy : 'appName';
+        const normalizedSortOrder =
+            sortOrder === 'asc' || sortOrder === 'desc' ? sortOrder : 'asc';
+
         const applications = await this.clientInstanceService.getApplications(
-            query,
+            {
```

Comment:
I'd extract the normalization as a method that returns this object. As a result the main controller method is not concerned with low level details